lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -337,7 +337,7 @@ fn sampling_limit(
 /// Determines the effective ef limit value for the given parameters.
 fn effective_limit(limit: usize, ef_limit: usize, poisson_sampling: usize) -> usize {
     // Prefer the highest of poisson_sampling/ef_limit, but never be higher than limit
-    poisson_sampling.max(ef_limit).min(limit)
+    poisson_sampling.clamp(limit, ef_limit)
```

Comment:
This is **not** the same, because `clamp` [panics](https://doc.rust-lang.org/std/primitive.f64.html#panics) if `min > max` which may happen in some configurations. The current with `max().min()` does not have this problem.

We discovered this the hard way because it panicked in production. It was reverted in <https://github.com/qdrant/qdrant/pull/1692>.